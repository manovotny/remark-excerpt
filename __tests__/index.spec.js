import {dirname, join} from 'node:path';
import {fileURLToPath} from 'node:url';

import {wrap} from 'jest-snapshot-serializer-raw';
import {remark} from 'remark';
import {toVFile} from 'to-vfile';

import plugin from '..';

const processFixture = async (name, options) => {
    const __filename = fileURLToPath(import.meta.url);
    const __dirname = dirname(__filename);
    const path = join(__dirname, 'fixtures', name);
    const file = await toVFile.read(path);
    const result = await remark().use(plugin, options).process(file);

    return result.toString();
};

test('should return unmodified document when no excerpt exists', async () => {
    const result = await processFixture('no-excerpt.md');

    expect(wrap(result)).toMatchSnapshot();
});

test('should return excerpt using "exceprt" identifier', async () => {
    const result = await processFixture('excerpt.md');

    expect(wrap(result)).toMatchSnapshot();
});

test('should return excerpt using "more" identifier', async () => {
    const result = await processFixture('more.md');

    expect(wrap(result)).toMatchSnapshot();
});

test('should return excerpt using "preview" identifier', async () => {
    const result = await processFixture('preview.md');

    expect(wrap(result)).toMatchSnapshot();
});

test('should return excerpt using "teaser" identifier', async () => {
    const result = await processFixture('teaser.md');

    expect(wrap(result)).toMatchSnapshot();
});

test('should return excerpt using custom identifier', async () => {
    const options = {
        identifier: 'custom',
    };
    const result = await processFixture('custom.md', options);

    expect(wrap(result)).toMatchSnapshot();
});

test('should return after excerpt when multiple exist', async () => {
    const result = await processFixture('multiple.md');

    expect(wrap(result)).toMatchSnapshot();
});

test('should handle identifier with dashes', async () => {
    const options = {
        identifier: 'custom-with--dashes',
    };
    const result = await processFixture('custom-with-dashes.md', options);

    expect(wrap(result)).toMatchSnapshot();
});

test('should handle identifier with spaces', async () => {
    const options = {
        identifier: 'custom with spaces',
    };
    const result = await processFixture('custom-with-spaces.md', options);

    expect(wrap(result)).toMatchSnapshot();
});

test('should handle no spacing in exceprt comment', async () => {
    const result = await processFixture('no-space-in-comment.md');

    expect(wrap(result)).toMatchSnapshot();
});

test('should handle mdx', async () => {
    const result = await processFixture('mdx.mdx');

    expect(wrap(result)).toMatchSnapshot();
});
