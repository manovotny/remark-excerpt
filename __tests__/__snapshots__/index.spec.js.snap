// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle identifier with dashes 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should handle identifier with spaces 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should handle mdx 1`] = `
import something from 'something';

export const meta = {
prop: 'value',
};

# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should handle no spacing in excerpt comment 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should return after excerpt when multiple exist 1`] = `
# Title

Paragraph 1.

`;

exports[`should return excerpt using "excerpt" identifier 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should return excerpt using "more" identifier 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should return excerpt using "preview" identifier 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should return excerpt using "teaser" identifier 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should return excerpt using custom identifier 1`] = `
# Title

Paragraph 1.

Paragraph 2.

`;

exports[`should return unmodified document when comment exists 1`] = `
# Title

Paragraph 1.

Paragraph 2.

<!-- comment -->

Paragraph 3.

Paragraph 4.

`;

exports[`should return unmodified document when no excerpt exists 1`] = `
# Title

Paragraph 1.

Paragraph 2.

Paragraph 3.

Paragraph 4.

`;
